Fix a warning message about the gce in-tree cloud provider state (#126773)
kubernetes/kubernetes: staging/src/k8s.io/cloud-provider/sample/README.md

1. Have your external repo under k8s.io. e.g. `k8s.io/cloud-provider-<provider>`
2. Create `main.go` file under your external repo CCM directory. Please refer to `basic_main.go` for a minimum working sample.
Note: If you have a requirement of adding/deleting controllers within CCM, please refer to `k8s.io/kubernetes/cmd/cloud-controller-manager/main.go` for extra details.

> 3. Build/release CCM from your external repo. For existing cloud providers, the option to import legacy providers from `k8s.io/legacy-cloud-provider/<provider>` is still available.

Author: When should `cmd/cloud-controller-manager` be removed from k/k?

Reviewer: we leave it as an example indefinitely @carlory

Author: From a code organization perspective, all the example projects live in the `staging` directory. Do we want to migrate and rename `cloud-controller-manager` to `staging/src/k8s.io/sample-cloud-controller-manager`? @dims

Reviewer 2: i would like to highlight that we had created a repository for a sample provider many years ago, but it never got populated. perhaps if we are going to create a staging directory for a sample we should link it to this repo https://github.com/kubernetes/cloud-provider-sample ?

> 3. Build/release CCM from your external repo. For existing cloud providers, the option to import legacy providers from `k8s.io/legacy-cloud-provider/<provider>` is still available until 1.31.

Reviewer 3: ```suggestion
3. Build/release CCM from your external repo. For existing cloud providers, the option to import legacy providers from `k8s.io/legacy-cloud-provider/<provider>` is not available since 1.31.
```
